### DISCUSSION TITLE
KAIST CS420: Compiler Design - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#643] [Announcements]
# KAIST CS420: Compiler Design - CS自学指南

CS自学指南

https://csdiy.wiki/%E7%BC%96%E8%AF%91%E5%8E%9F%E7%90%86/CS420/

**Comment by Qi-Zhan** [comment]
确实是好课, 代码量很大, 最终实现的 C 语言编译器应该是所有课程里面最强大的了, 个人觉得特别适合在学完编译原理之后再肝, 不然可能有点难了

**Reply by wyxxxcat** [reply]
这个课程相关代码和文章能公开吗

**Reply by Qi-Zhan** [reply]
你说的是实现的代码吗, 肯定不应该公开的. 框架代码在 https://github.com/kaist-cp/kecc-public.
不太清楚你说的文章是什么

**Reply by wyxxxcat** [reply]
就是我觉得这个课程很好，我想写一些包含课程实现代码的笔记分享

**Reply by ghost** [reply]
我想问一下，QIzhan先生，你说的实现C语言编译器是哪个课程？我准备先学编译原理再学这个。

**Reply by Qi-Zhan** [reply]
就是这个 CS420 实现的编译器呀, 我的意思是最好有一门偏理论的课作为基础(比如自己学校的编译原理), 然后再来学 cs420, 因为这个课理论部分相当少, 直接学可能不太适应

**Reply by guojianchen** [reply]
本来就特别难

**Comment by HanfengLiao** [comment]
求一个Robotics学习路线

**Reply by AURORALLLL** [reply]
本科毕业了，想写一个，但是感觉好难啊

**Reply by Qi-Zhan** [reply]
是挺难的, 尤其是 lab2 代码生成, 即使他视频里面有很多代码可以用还是很难.
不过坚持写下来收获也是相当大的

**Reply by AzenAnn** [reply]
我也想搞一个 q2020779856

**Comment by AzenAnn** [comment]
caigejiaoyin

**Comment by zongwei-tang** [comment]
Amazing!

**Comment by kaizjh** [comment]
看完了，太棒啦，作者是两三年学了这么多吗（计算机系的几大核心课程：体系、网络、操统、编译）？太厉害啦！！！非常感谢作者的分享，受益匪浅。

**Comment by lee23james** [comment]
请问这里面有计算机组成原理的相关好课吗？

**Reply by Qi-Zhan** [reply]
cs61c

**Reply by lee23james** [reply]
谢谢，准备学习了

**Comment by nonhana** [comment]
受朋友推荐，前来还技术债了

**Comment by zerorobot0** [comment]
想向路过的各位大佬请教下CS61A 课程网站为什么点不开呀？有没有跳过登陆直接进入伯克利内学习网站的方法呀？万分感谢！

**Reply by zongwei-tang** [reply]
https://github.com/Stromxun/CS_learning，这个链接有课程作业之类的

**Comment by Ku4nYu** [comment]
Greatest and legendary book before the age of 20s for those who major in CS.

**Comment by GhostPG01** [comment]
很多链接失效了，希望可以更新补档一下

**Comment by zerorobot0** [comment]
之前的评论怎么没有啦 真的很需要呜呜

**Comment by backpocket7** [comment]
请问能不能开一个或者几个discord groups, for Q&A and discussions.
The main group will cover high level discussions.
Each group will cover specific class.

**Comment by Vanilla-Beauty** [comment]
有没有想一起学CS420的小伙伴啊, 欢迎 QQ: 807077266 交流

**Reply by RockyXYB** [reply]
It's very good.

**Comment by LIHUA919** [comment]
这是好事呀

**Comment by mateThinker** [comment]
芝麻怎么用不了了，有人能私发个能用的吗，非常感谢

**Reply by mateThinker** [reply]
已解决，谢谢

**Comment by A-Handsomeman** [comment]
Cs61a秋季的代码有可用的吗

**Comment by thxcn** [comment]
现在看YouTube视频需要berkeley的账号哎

**Reply by A-Handsomeman** [reply]
我去看了下，cs61a里的链接点进去是John DeNero教授账号上的视频，这里面的没有要求，方便把需要账号的视频发一下吗

**Reply by thxcn** [reply]
https://cs61a.org/lab/lab06/

https://youtu.be/playlist?list=PLx38hZJ5RLZfrzyQvcEjb9j4RWMJRVbUB

**Reply by thxcn** [reply]
我看是Recording需要账号登录，Videos则不需要🤧

**Comment by AkunCube** [comment]
新建了一个kecc qq群，大家感兴趣可以进群讨论学习：1012716373
